Compiling fails on FreeBSD

```
% cargo build
   Compiling pprof v0.10.0 (/usr/home/dveeden/dev/pprof-rs)
error[E0425]: cannot find function `create_pipe` in this scope
  --> src/addr_validate.rs:52:35
   |
52 |         let (read_fd, write_fd) = create_pipe()?;
   |                                   ^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find value `addr` in this scope
   --> src/profiler.rs:280:44
    |
280 |                 if profiler.is_blocklisted(addr) {
    |                                            ^^^^ not found in this scope

error[E0308]: mismatched types
   --> src/profiler.rs:205:9
    |
205 | /         unsafe {
206 | |             #[cfg(target_os = "linux")]
207 | |             {
208 | |                 let errno = *libc::__errno_location();
...   |
215 | |             }
216 | |         }
    | |_________^ expected struct `ErrnoProtector`, found `()`

Some errors have detailed explanations: E0308, E0425.
For more information about an error, try `rustc --explain E0308`.
error: could not compile `pprof` due to 3 previous errors
```

```
% freebsd-version 
13.0-RELEASE-p11
% cargo --version
cargo 1.63.0-nightly
```

--- a/src/addr_validate.rs
+++ b/src/addr_validate.rs
@@ -28,7 +28,7 @@ fn create_pipe() -> nix::Result<(i32, i32)> {
 }
 
 #[inline]
-#[cfg(target_os = "macos")]
+#[cfg(any(target_os = "macos", target_os = "freebsd"))]
 fn create_pipe() -> nix::Result<(i32, i32)> {
     use nix::fcntl::{fcntl, FcntlArg, FdFlag, OFlag};
     use nix::unistd::pipe;

--- a/src/profiler.rs
+++ b/src/profiler.rs
@@ -243,7 +243,7 @@ impl ErrnoProtector {
                 let errno = *libc::__errno_location();
                 Self(errno)
             }
-            #[cfg(target_os = "macos")]
+            #[cfg(any(target_os = "macos", target_os = "freebsd"))]
             {
                 let errno = *libc::__error();
                 Self(errno)
@@ -263,7 +263,7 @@ impl Drop for ErrnoProtector {
             {
                 *libc::__errno_location() = self.0;
             }
-            #[cfg(target_os = "macos")]
+            #[cfg(any(target_os = "macos", target_os = "freebsd"))]
             {
                 *libc::__error() = self.0;
             }
@@ -303,6 +303,9 @@ extern "C" fn perf_signal_handler(
                 let addr =
                     unsafe { (*ucontext).uc_mcontext.gregs[libc::REG_RIP as usize] as usize };
 
+                #[cfg(all(target_arch = "x86_64", target_os = "freebsd"))]
+                let addr = unsafe { (*ucontext).uc_mcontext.mc_rip as usize };
+
                 #[cfg(all(target_arch = "x86_64", target_os = "macos"))]
                 let addr = unsafe {
                     let mcontext = (*ucontext).uc_mcontext;
@@ -318,6 +321,9 @@ extern "C" fn perf_signal_handler(
                     any(target_os = "android", target_os = "linux")
                 ))]
                 let addr = unsafe { (*ucontext).uc_mcontext.pc as usize };
+
+                #[cfg(all(target_arch = "aarch64", target_os = "freebsd"))]
+                let addr = unsafe { (*ucontext).mc_gpregs.gp_elr as usize };
 
                 #[cfg(all(target_arch = "aarch64", target_os = "macos"))]
                 let addr = unsafe {

